GitHubCore.ps1

```diff
@@ -130,6 +148,32 @@ function Invoke-GHRestMethod
 
     Invoke-UpdateCheck
 
+    # Minor error checking around $InFile
+    if ((-not [String]::IsNullOrWhiteSpace($InFile)) -and ($Method -ne 'Post'))
```

Comment:
That's fair.  I think my memory had flaked here on whether that could be used successfully on an optional parameter.  I just verified that it would work fine.  Will change.

```powershell
function Invoke-NotNull {
    [cmdletbinding()]
    param(
        [ValidateNotNullOrEmpty()]
        [string] $Value
    )

    write-host "[$Value]"
}

Invoke-NotNull
[]
Invoke-NotNull -Value 'foo'
[foo]
Invoke-NotNull -Value $null
Invoke-NotNull : Cannot validate argument on parameter 'Value'.
```

Reply:
The scenario I have specifically in my head is something like:

```powershell
New-GitHubRelease ... -Label $myLabel
```

in a case where I forgot to set `$myLabel`

Reply:
> in a case where I forgot to set `$myLabel`

There is no `Label` for that function, so I'll assume your example was supposed to be `-Description $myDescription`.

Ah, so this is where it gets tricky.  It's valid to have no description.  I consider it a _feature_ that earlier logic in your script might have determined that the description was `$null` and now you're just blindly passing it in.  We'll take that input and just ignore it because `[String]::IsNullOrEmpty`.  Is there a more consistent PowerShell philosophy in existing cmdlets where it doesn't work that way?

Reply:
> Ah, so this is where it gets tricky. It's valid to have no description. I consider it a feature that earlier logic in your script might have determined that the description was $null and now you're just blindly passing it in.

Yeah, that's exactly why I'd use `[ValidateNotNullOrEmpty()]` on a non-mandatory parameter, to get this behaviour:

```
> function Test { param([ValidateNotNullOrEmpty()][Parameter()]$x) $x }

> Test

> Test -x $notSet
Test: Cannot validate argument on parameter 'x'. The argument is null or empty. Provide an argument that is not null or empty, and then try the command again.

> Test -x banana
banana
```

This way, it's perfectly valid to not specify the parameter, but if you do specify it, it shouldn't be the same as not specifying it.

The only issue there is that that means that more automated uses where the parameter is constructed, for example with splatting, have to be slightly more careful, since having a default value of null or empty string will cause a parameter binding failure.

Reply:
Anyway, point is that it's totally philosophical, but just wanted to make sure it's a deliberate choice

Reply:
> The only issue there is that that means that more automated uses where the parameter is constructed, for example with splatting, have to be slightly more careful, since having a default value of null or empty string will cause a parameter binding failure.

That's the point I was trying to drive at.  It _seems_ like it's a positive feature to allow `$null` to be sent in and thus have the parameter ignored.  But I'm just a single user, and so I _am_ interested in hearing from others if they feel similarly or not.

Reply:
Yeah, I think that's a totally valid point.

My personal usage, especially of a module like this, is a lot more interactive, and tends to involve something like executing lines that I've already written. But if I get them out of order and it quietly goes ahead, I have to go and fix it on GitHub (perhaps the lesson is that I should use `-WhatIf`...).

I don't feel that strongly about it. My real 2 cents are that it's less of a breaking change to turn it from an error into something that works later than to go the other way.